lib/redix_sentinel.ex

```diff
@@ -317,4 +318,10 @@ defmodule RedixSentinel do
       end
     end)
   end
+
+  defp confirm_role!(conn, role) do
+    [^role | _] = Redix.command!(conn, ["ROLE"])
+    rescue Redix.Error ->
+      true = Redix.command!(conn, ["INFO", "replication"]) =~ "role:#{role}"
```

Comment:
I would also prefer we parse the response instead of resorting to regex here.

Reply:
Yes and No. Yes this doesn't buy us much over the regex match, but I try to avoid regex because of the false positives. Sometimes it will match in an unexpected way. Given that parsing the response is not that complicated, I prefer it.